### PR TITLE
New what is devops article

### PR DESCRIPTION
**Pull request by interurban**
## Description
Adding a what is devops page

### CODE REVIEW

File: themes/default/content/what-is/what-is-devops.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+- **Continuous integration and continuous delivery (CI/CD)**: Continuous integration (CI) and continuous delivery (CD) are interconnected practices that streamline the software development process. CI involves developers frequently merging code changes into a central repository where automated builds and tests are conducted to identify and fix bugs swiftly, thus enhancing software quality and shortening the update validation period. CD builds on CI's foundation, automatically building, testing, and preparing code for release. It further automates the deployment of these changes to various testing or production environments, ensuring that the code is not just ready for deployment but is continuously delivered to production with minimal manual intervention. This creates a deployment-ready build artifact that has been rigorously tested, facilitating frequent and stable releases, with monitoring systems in place to ensure transparency throughout the process. [Learn more about Pulumi’s CI/CD integration](https://www.pulumi.com/docs/guides/continuous-delivery/).
```
Comment (aaronkao) [inline themes/default/content/what-is/what-is-devops.md:21]: add the link to the CICD page Diana is creating and ship it together

Comment (aaronkao) [reply themes/default/content/what-is/what-is-devops.md:21]: make sure the definitions are aligned with each other and not too far off

File: themes/default/content/what-is/what-is-devops.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+- **Testing infrastructure as code**: [Testing infrastructure as code](https://www.pulumi.com/docs/using-pulumi/testing/) (IaC) in DevOps is similar to testing applications using standard testing tools for various test types. Unit tests swiftly validate code segments without real infrastructure, relying on the same programming language as the IaC programs. Property tests take a step further by provisioning real infrastructure to check resource configurations before and after deployment. Integration tests comprehensively establish a temporary infrastructure setup to verify end-to-end functionality, particularly useful in CI/CD workflows. For more robust testing, one might employ fuzz testing to challenge the infrastructure with unexpected inputs, chaos testing to assess resilience by intentionally causing disruptions, and stress testing to push the system to its limits. These advanced tests, enabled by the flexibility of using general-purpose languages for IaC, provide a wide array of tools to ensure the infrastructure's integrity and robustness.
```
Comment (aaronkao) [inline themes/default/content/what-is/what-is-devops.md:25]: this is actually different from the other bullets where you put the Pulumi related article as part of the Learn more

File: themes/default/content/what-is/what-is-devops.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+| Continuous testing         | The process of executing automated tests as part of the software delivery pipeline to obtain immediate feedback on the business risks associated with a software release candidate. |
+| Continuous deployment      | An extension of continuous delivery, where every change that passes the automated tests is deployed to production automatically, without explicit approval from a developer, making the deployment process fully automated. |
+| Version control            | A system that records changes to a file or set of files over time so that you can recall specific versions later. It's a fundamental tool for enabling collaboration, ensuring traceability, and managing code for teams. |
+| Agile methodology          | A set of principles for software development under which requirements and solutions evolve through the collaborative effort of cross-functional teams. Agile advocates adaptive planning, evolutionary development, early delivery, and continual improvement, and it encourages flexible responses to change. |
+
+## DevOps tools
```
Comment (scottslowe) [inline themes/default/content/what-is/what-is-devops.md:45]: It feels like this section should list some popular DevOps tools that readers might recognize, as well as have the text about how Pulumi ties into the various DevOps areas.

**Review by scottslowe — APPROVED** [review]
Just one suggestion/thought for consideration; otherwise, looks good.

**Pull request merged**